tests/x509_openssl_ut/x509_openssl_ut.c

```diff
@@ -368,7 +368,7 @@ BEGIN_TEST_SUITE(x509_openssl_unittests)
         //arrange
 
         //act
-        int result = x509_openssl_add_credentials(NULL, "certificate", "privatekey");
+        int result = x509_openssl_add_credentials(NULL, "certificate", "privatekey", NULL);
```

Comment:
>NULL [](start = 85, length = 4)

Require UT for new option

Reply:
It looks like there is a requirement that `If any argument is NULL then x509_openssl_add_credentials shall fail and return a non-zero value.`. My design was to allow user to set either third or fourth argument to NULL if they use plaintext vs cryptodev private key.

Do you believe it would be better to introduce a completely new API call (e.g. `x509_openssl_app_cryptodev_key`)?

Reply:
Yes - I would be interested in seeing a different method in the API design for the private keys. We cannot break the existing "API contract" for backwards-compat reasons.

Reply:
@massand Updated the PR to keep the old API untouched. So to add unit tests got the new call (`x509_openssl_add_credentials_cryptodev`) do we need to create the requirements somewhere, or I can just add them right away (adapt from those for `x509_openssl_add_credentials` probably)?

Reply:
Yes, please add the requirements adapted from an existing module (such as x509_openssl_add_credentials). I try to keep the requirements to a minimum of happy path requirements and requirements specifying major error conditions. I would also keep requirements scoped to public interface API methods, instead of each internal method. Thank you for maintaining the consistency again!

Reply:
@massand Sorry, I've got a bit stuck with debugging the umock. So I extended `setup_load_credentials` to support crypto device scenario. My test crashes, and debugging shows it's because some of `SetReturn` calls are being ignored. As far as I understand it can happen if the sequence of `STRICT_EXPECTED_CALL`s is incorrect, so umock can't identify `SetReturn` with this particular call.
So how can I debug this issue further? How can I see if umock has 'checked' a `STRICT_EXPECTED_CALL` I've defined?

Reply:
Added tests, please review.

Only tested on Linux with `OPENSSL_VERSION_NUMBER == 0x1010007f`, but I _tried_ to get the test numbering right.